packages/perennial/contracts/multiinvoker/MultiInvoker.sol

```diff
@@ -208,11 +209,13 @@ contract MultiInvoker is IMultiInvoker, UInitializable {
      * @param amount Amount of DSU to unwrap
      */
     function _unwrap(address receiver, UFixed18 amount) private {
-        // Unwrap the DSU into USDC and return to the receiver
-        // The current batcher does not have UNWRAP functionality yet, so just go directly to the reserve
-        reserve.redeem(amount);
-
-        // Push the amount to the receiver
-        USDC.push(receiver, amount);
+        // If the batcher doesn't have enough for this unwrap, go directly to the reserve
+        if (amount.gt(USDC.balanceOf(address(batcher)))) {
+            reserve.redeem(amount);
+            if (receiver != address(this)) USDC.push(receiver, amount);
```

Comment:
we don't need this check here I think since the `_unwrap()` call happens after the withdrawal in `unwrapAndWithdraw()`

Reply:
relatedly, should we block `receiver` from being equal to `address(this)` in `wrap` and `unwrap`?

Reply:
I don't think we should try to protect the user too much here, unless we feel that this could be dangerous for some reason. Also, I like keeping this line in as an for symmetry and extra protection in case we use this code in some other way in future impls